packages/cardpay-sdk/sdk/reward-pool/base.ts

```diff
@@ -513,28 +534,36 @@ The reward program ${rewardProgramId} has balance equals ${fromWei(
   }
 
   async claimAll(
-    unclaimedValidProofs: WithSymbol<ClaimableProof>[],
-    accountAddress?: string
+    safeAddress: string,
+    rewardProgramId?: string,
+    tokenAddress?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
   ): Promise<SuccessfulTransactionReceipt[]> {
-    if (unclaimedValidProofs.length == 0) {
-      throw new Error('Do not have any valid proofs to claim');
-    }
     let rewardManager = await getSDK('RewardManager', this.layer2Web3);
-    const safeOwner = accountAddress ?? (await rewardManager.getRewardSafeOwner(unclaimedValidProofs[0].safeAddress));
-    const proofs = unclaimedValidProofs
-      .filter((o) => o.isValid)
-      .filter((o) => {
-        //Intentionally step over claims which are not big enough to pay for gas fees
-        return o.gasEstimate.amount.lt(new BN(o.amount));
-      });
+    let rewardSafeOwner = await rewardManager.getRewardSafeOwner(safeAddress);
+    const unclaimedValidProofs = (
+      await this.getProofs(rewardSafeOwner, safeAddress, rewardProgramId, tokenAddress, false)
+    ).filter((o) => o.isValid);
+    const unclaimedValidProofsWithoutCryptoDust = unclaimedValidProofs.filter((o) => {
+      return o.gasEstimate.amount.lt(new BN(o.amount));
+    });
+    console.log(
+      `Claiming ${unclaimedValidProofsWithoutCryptoDust.length} out of ${unclaimedValidProofs.length} proofs`
+    );
+    return this.claimAllProofs(unclaimedValidProofsWithoutCryptoDust, safeAddress, txnOptions, contractOptions);
+  }
+
+  async claimAllProofs(
```

Comment:
nit:  Maybe this could be called `claimProofs`,  to avoid confusing with claiming all available proofs, when I first read it I got confused on the different of `claimAll`  and `claimAllProofs`

Reply:
Yea actually this makes a lot of sense. 

When using "All" it seems to mean **claim what you can**.  `claimAllProofs` doesn't make the assumption that mean to **claim what you can**, so you can try claiming a proof which has crypto dust too.